mlrun/track/base_tracker.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+from abc import abstractmethod
+
+import mlrun
+from mlrun.execution import MLClientCtx
+
+from ..utils import logger
+from .tracker import Tracker
+
+
+class BaseTracker(Tracker):
+    """
+    a base class for specific trackers, starts to implement class functions
+    """
+
+    # The module name this tracker will look for.
+    TRACKED_MODULE_NAME = ...
+
+    def __init__(self):
+        super().__init__()
+        self._tracked_platform = importlib.import_module(self.TRACKED_MODULE_NAME)
+        self._run_track_kwargs = {}
+        # will add artifacts in mlrun format to here
+        self._artifacts = {}
+
+    @classmethod
+    def is_enabled(cls) -> bool:
+        # Check the tracker's configuration:
+        if getattr(mlrun.mlconf.tracking, cls.TRACKED_MODULE_NAME).mode != "enabled":
```

Comment:
I mean that each tracking class should have its "name" which is also what the config uses for it - for example in your case it would be `mlflow`. You can use it here to check if the class is enabled or not.
Then each class will have a function it needs to override to load needed packages and prepare for action, which may be called from `init` or whatever flow makes sense. This method may `import` multiple packages, may initialize a client code if needed and so on and so forth. However, it should not depend on the name of the tracking class. Specifically for your case, it seems that all you need in your case is to `import mlflow`, which is a simple case of this flow.

Reply:
done